tools/spm/generate_partition_code.py

```diff
@@ -0,0 +1,439 @@
+#!/usr/bin/python
+import glob
+import itertools
+import json
+import os
+from os.path import join as path_join
+
+from jinja2 import Environment, FileSystemLoader
+from jsonschema import validate
+
+SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+TEMPLATES_DIR = path_join(SCRIPT_DIR, 'templates')
+
+
+def assert_int(num):
+    """
+    Tries to parse an integer num from a given string
+    
+    :param num: Number in int/string type
+    :return: Numeric value
+    """
+    if isinstance(num, int):
+        return num
+    num_str = str(num)
+    radix = 16 if num_str.lower().startswith('0x') else 10
+    res = int(num_str, radix)
```

Comment:
It looks like you're implementing python's "base 0". the [`int`](https://docs.python.org/3/library/functions.html?highlight=int#int) function will "intemperate numbers as code literals" when `radix=0`. Using that, the prior two lines could be:
```python
res = int(num_str, 0)
```